pkg/jobmanager/jobmanager.go

```diff
@@ -141,6 +141,8 @@ func (jm *JobManager) Run(ctx xcontext.Context, resumeJobs bool) error {
 		return fmt.Errorf("Cannot start API: %w", err)
 	}
 
+	ctx = ctx.WithTag("serverID", a.ServerID())
```

Comment:
should be done in main to capture all events emitted by an instance

Reply:
Unfortunately, the final ServerID value is constructed first only on `api.New`. So either I need to move around related logic, or add this later.

Reply:
yes, please pull out the "flag or hostname" logic out of api so that it is determined before API is instantiated.